### PR TITLE
Add Maty's GameTest gist into Neo repo

### PR DESCRIPTION
**Pull request by TelepathicGrunt**
Original gists: 
https://gist.github.com/Matyrobbrt/832c0d88a90ae15f5f665bb794a48198
https://gist.github.com/Matyrobbrt/cac2427748a51ff2bfc82fcd03606b55

Someone should write one for JUnit tests as well for balance, as all things should be

Make sure this is merged with no build made somehow

### CODE REVIEW

**Review by shartte — DISMISSED** [review]

File: docs/MCGAMETESTS.md
Comment (Matyrobbrt) [inline docs/MCGAMETESTS.md:1]: The name of this file is a tad misleading, I'd name it `TESTFRAMEWORK`.

File: docs/NEOGAMETESTS.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+@TestHolder(description = "Does something")
+static void gameTestOnly(final ExtendedGameTestHelper helper) {
+    // ....
+    helper.succeed();
+}
+```
+
+## Extra: method-based event tests
+These tests cannot be automated via gametests, and will only provide manual event testing. They should primarily be used by client tests.  
+More information can be found [here](MCGAMETESTS.md).
```
Comment (Matyrobbrt) [inline docs/NEOGAMETESTS.md]:
```suggestion
More information can be found [here](MCGAMETESTS.md).
If you're unsure how to write a test for a feature, feel free to ask in our [Discord](https://discord.neoforged.net/) server, in the [`#neoforge-github` channel](https://discord.com/channels/313125603924639766/852298000042164244), or consult the existing tests.
```

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]

**Comment by Matyrobbrt** [comment]
Please also link the second file in the `CONTRIBUTING.md` file under section 8, a subsection that says something along the lines of "Write an (automated) test under the `tests` project, if possible".

**Comment by Matyrobbrt** [comment]
Oh also, since this is a documentation update PR, can you also update the readme to mention 1.21 as active support?

File: docs/MCGAMETESTS.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+   
+  .allowClientModifications() // Allow OP'd clients to modify the status of tests, and to enable them
+  .syncToClients() // Sync the status of tests to clients
+
+  .build().create(); // Build and store the InternalTestFramework. We use the "internal" version because we want to access methods not usually exposed, like the init method
+  
+  // Initialise this framework, using the mod event bus of the currently loading mod, and the container of the currently loading mod.
+  // The container is used for collecting annotations.
+  // This method will collect and register tests, structure templates, group data, and will fire init listeners.
+  framework.init(FMLJavaModLoadingContext.get().getModEventBus(), ModLoadingContext.get().getActiveContainer());
```
Comment (Spinoscythe) [inline docs/MCGAMETESTS.md]: Remove the reference to FMLJavaModLoadingContext

**Review by Spinoscythe — CHANGES_REQUESTED** [review]

**Comment by HenryLoenwind** [comment]
BTW, has someone cross-checked this against https://docs.neoforged.net/docs/misc/gametest/ ? Are they equivalent? Is this a superset (then the website docs should mention this one as further reading)?

**Comment by TelepathicGrunt** [comment]
@HenryLoenwind Pretend you are making a NeoForge PR and you need to write a gametest. Now look at the Game Test doc. Then the gametest file in this PR. Which one is going to set you up better for writing a game test?

**Comment by HenryLoenwind** [comment]
> @HenryLoenwind Pretend you are making a NeoForge PR and you need to write a gametest. Now look at the Game Test doc. Then the gametest file in this PR. Which one is going to set you up better for writing a game test?

Clearly the doc. It's the only one of the two that actually tells me how to run the tests.

But that was not my question. The question was if there's anything in this one that is generic and should go into the docs instead to avoid repetition---and here I'm assuming stuff in this one doesn't exist for mods and only neo's dev env, which is something that's not clear from taking a quick look at this one either.

So again, has someone who actually understands this stuff cross-checked this with the docs to eliminate duplications? Is there something in here that would better go into the docs or something that would better reference the docs and explain the differences instead of repeating them?

**Comment by TelepathicGrunt** [comment]
I find the file here far easier to follow than the docs. The docs broke up the examples too much into piecewise and I can't follow them any more and have no idea how any of the gametest is put together. The example here is more complete and actually shows the context of how it is put together. That is my opinion.

**Review by Matyrobbrt — APPROVED** [review]

**Pull request merged**